packages/graphiql-toolkit/src/async-helpers/index.ts

```diff
@@ -52,42 +52,34 @@ export function isAsyncIterable(
   );
 }
 
-function asyncIterableToPromise<T>(
+async function asyncIterableToPromise<T>(
   input: AsyncIterable<T> | AsyncIterableIterator<T>,
 ): Promise<T> {
-  return new Promise((resolve, reject) => {
-    // Also support AsyncGenerator on Safari iOS.
-    // As mentioned in the isAsyncIterable function there is no Symbol.asyncIterator available
-    // so every AsyncIterable must be implemented using AsyncGenerator.
-    const iteratorReturn = (
-      'return' in input ? input : input[Symbol.asyncIterator]()
-    ).return?.bind(input);
-    const iteratorNext = (
-      'next' in input ? input : input[Symbol.asyncIterator]()
-    ).next.bind(input);
+  // Also support AsyncGenerator on Safari iOS.
+  // As mentioned in the isAsyncIterable function there is no Symbol.asyncIterator available
+  // so every AsyncIterable must be implemented using AsyncGenerator.
+  const iteratorReturn = (
+    'return' in input ? input : input[Symbol.asyncIterator]()
+  ).return?.bind(input);
+  const iteratorNext = (
+    'next' in input ? input : input[Symbol.asyncIterator]()
+  ).next.bind(input);
 
-    iteratorNext()
-      .then(result => {
-        resolve(result.value);
-        // ensure cleanup
-        iteratorReturn?.();
-      })
-      .catch(err => {
-        reject(err);
-      });
-  });
+  const result = await iteratorNext();
+  // ensure cleanup
+  iteratorReturn?.();
+  return result.value;
```

Comment:
the only thing is that after refactoring to `async/await` we call `iteratorReturn?.()` before returning values, previously we did following, does it ok?
```js
        resolve(result.value);
        // ensure cleanup
        iteratorReturn?.();
```